x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/DataStreamDeprecationCheckerTests.java

```diff
@@ -284,10 +285,11 @@ public void testOldIndicesIgnoredWarningCheck() {
 
         DeprecationIssue expected = new DeprecationIssue(
             DeprecationIssue.Level.WARNING,
-            "Old data stream with a compatibility version < 9.0 Have Been Ignored",
-            "https://www.elastic.co/guide/en/elasticsearch/reference/master/breaking-changes-9.0.html",
-            "This data stream has read only backing indices that were created before Elasticsearch 9.0.0 and have been marked as "
-                + "OK to remain read-only after upgrade",
+            "Old data stream with a compatibility version < " + Version.CURRENT.major + "0.0 Has Been Ignored",
+            "https://ela.st/es-deprecation-ds-reindex",
+            "This data stream has read only backing indices that were created before Elasticsearch "
+                + Version.CURRENT.major
+                + "0.0 and have been marked as OK to remain read-only after upgrade",
```

Comment:
Nit: When we check the lucene compatibility for indices we hardcode version `9.0.0` but when we check it for data streams it's not version specific. I am wondering why we have different approaches in these two cases; I initially expected them to be treated in the same way.

Reply:
I see, I think we own both so if this is stopping you then I understand. I think Kibana should be ok, I do not think it parses the messages, if you have time we could double check it with them; otherwise, we can leave it as it is.

Reply:
Ahhh good stuff. I thought core owned the index deprecation notices... Anyway I have updated those to now use a dynamic version number but was careful to make sure the messages didn't actually change their content to sidestep any kibana issues.